src/Framework/MSBuildEventSource.cs

```diff
@@ -0,0 +1,338 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+
+
+//
+// This captures information of how various key methods of building with MSBuild ran.
+//
+namespace Microsoft.Build.Eventing
+{
+    // /OnlyProviders=*MSBuild-Profiling
+    [EventSource(Name = "MSBuild-Profiling")]
```

Comment:
Nit: I would just call this "Microsoft-Build" to match the standard naming conventions that are used across products.

Reply:
I'd also remove "-Profiling" as well.  This is more generically useful than just for profiling.